src/components/ChallengeModal/ChallengeModal.tsx

```diff
@@ -55,6 +62,33 @@ interface ChallengeModalProperties {
     playersList?: Array<{ name: string; rank: number }>;
     tournamentRecordId?: number;
     tournamentRecordRoundId?: number;
+    created?: (c: ChallengeDetails) => void;
+}
+
+interface ChallengeState {
+    live?: boolean; // computed during creation of challenge
+    initialized: boolean;
+    min_ranking: number;
+    max_ranking: number;
+    challenger_color: string; // "automatic"
+    game: {
+        name: string; //"",
+        rules: string; // "japanese",
+        ranked: boolean;
+        width: IdType;
```

Comment:
Not sure I'd use `IdType` here.  While the underlying `number | string` is correct, semantically `IdType` says "this is an ID", right?

Reply:
True, I originally had number | string, and plonked that in as a last minute "oh, we have this type".

I actually don't even like that it has to be number or string, so that's another darn thing to fix